src/pushsource/_impl/model/azure.py

```diff
@@ -0,0 +1,60 @@
+import re
+
+from frozenlist2 import frozenlist
+
+from .. import compat_attr as attr
+from .conv import in_, instance_of, optional_str, optional
+from .vms import VMIPushItem
+
+
+VHDGeneration = ["V1", "V2"]
+
+
+@attr.s()
+class VHDPushItem(VMIPushItem):
+    """A :class:`~pushsource.PushItem` representing a Virtual Hard Disk (or VHD).
+
+    For push items of this type, the :meth:`~pushsource.PushItem.src` attribute
+    refers to a file containing the VHD file..
+
+    This library does not verify that the referenced file is a valid
+    snapshot.
+
+    Note: As some attributes can not be obtained directly from Koji they're marked
+    as optional in the `attrs` validators. With this the VHDPushItem objects can be
+    created without these attributes and the missing values can be added later by
+    using `attrs.evolve`.
+    """
+
+    generation = attr.ib(type=str, default="V2", validator=optional(in_(VHDGeneration)))
+    """VHD generation. It can be ``V1`` or ``V2``."""
+
+    sku_id = attr.ib(type=str, default=None, validator=optional_str)
+    """SKU ID for the VM image generation, generally being the Plan ID."""
+
+    support_legacy = attr.ib(type=bool, default=False, validator=instance_of(bool))
+    """If the image ``generation == V2`` but also supports ``V1``."""
+
+    legacy_sku_id = attr.ib(
+        type=str, default=None, validator=optional(in_(VHDGeneration))
+    )
+    """The ``V1`` SKU ID (only if ``support_legacy`` is true)."""
+
+    disk_version = attr.ib(type=str, default=None)
+    """The image's disk version according to Azure's format {int}.{int}.{int}"""
+
+    recommended_sizes = attr.ib(
```

Comment:
Does order matter for the elements in this list?  For example is the first size "more recommended" than the next?

If order doesn't matter, you might consider making the converter sort() them so that there aren't multiple possible representations of the "same" value.

Reply:
It doesn't matter. Sure, I'll add the converter, thanks!

Reply:
@rohanpm I've added a sorting for it, please let me know if it's good now